vuu-ui/packages/vuu-shell/src/application-settings/ApplicationSettingsPanel.tsx

```diff
@@ -1,58 +1,193 @@
+              );
+            } else {
+              return (
+                <Option
+                  value={value}
+                  key={value}
+                  data-field={property.name}
+                ></Option>
+              );
+            }
+          })}
+        </Dropdown>
+      );
+    }
+  } else {
+    return <Input></Input>;
+  }
+}
 
-export interface ApplicationSettingsPanelProps
+// Props for Panel
+export interface ApplicatonSettingsPanelProps
   extends HTMLAttributes<HTMLDivElement> {
-  settings: { [key: string]: unknown };
+  applicationSettingsSchema: SettingsSchema;
+  applicationSettings: Record<string, string | number | boolean>;
+  onApplicationSettingChanged: (
+    propertyName: string,
+    value: string | number | boolean
+  ) => void;
 }
 
+// Gets the field form element for a dropdown selection box for selection change handler
+export const queryFieldFormElement = <T extends HTMLElement = HTMLElement>(
+  el: HTMLElement | EventTarget | null
+) => {
+  if (el === null) {
+    return null;
+  }
+  const dataFieldValue = el.getAttribute("data-field");
+  const saltFormField = document.querySelector(".saltFormField") as HTMLElement;
+  if (saltFormField && saltFormField.dataset.field) {
+    return document.querySelector(`[data-field="${dataFieldValue}"]`);
+  }
+};
+
+// Generates application settings form component
+export const SettingsForm = ({
+  applicationSettingsSchema,
+  applicationSettings,
+  onApplicationSettingChanged,
+}: ApplicatonSettingsPanelProps) => {
+  // Settings Handler for toggle and input buttons
+  const onSettingChanged = useCallback<FormEventHandler>((event) => {
+    const fieldElement = queryClosest(event.target, "[data-field]");
+    const fieldName = getFieldName(fieldElement);
+    onApplicationSettingChanged(fieldName, event.target.value);
+  }, []);
+
+  // Seperate change handler for selection form controls (to be implemented)
+  const handleSelectionChange = useCallback<SingleSelectionHandler>(
+    (event, selected) => {
+      console.log(event.target);
+      const fieldElement = queryFieldFormElement(event.target);
+      const fieldName = getFieldName(fieldElement);
+      onApplicationSettingChanged(fieldName, selected);
+    },
+    []
+  );
+
+  return (
+    <div>
+      {applicationSettingsSchema.properties.map((property) => (
+        <FormField data-field={property.name} key={property.name}>
+          <FormFieldLabel>{property.label}</FormFieldLabel>
+          {getFormControl(
+            property,
+            onSettingChanged,
+            handleSelectionChange,
+            applicationSettings[property.name]
+          )}
+        </FormField>
+      ))}
+    </div>
+  );
+};
+
 const classBase = "vuuApplicationSettingsPanel";
 
-export const ApplicationSettingsPanel = () => {
+export const ApplicationSettingsPanel = ({
+  applicationSettingsSchema,
+  applicationSettings,
+  onApplicationSettingChanged,
+}: ApplicatonSettingsPanelProps) => {
```

Comment:
our props extends A DIVElement (HTMLAttributes<HTMLDivElement>)

that means we potentially have a bunch of additional props here , which we're ignoring. 
We know nthese are all props of a div element so we can spread them stright onto our div.

I like to call the spread prop for this `htmlAttributes`, so it would look like this
```
export const ApplicationSettingsPanel = ({
  applicationSettingsSchema,
  applicationSettings,
  onApplicationSettingChanged,
...htmlAttributes
}: ApplicatonSettingsPanelProps) => {
```
and then you would apply these attributes to the outer div container like this

```
{...htmlAttributes}
```